src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1387,17 +1403,27 @@ private function dumpValue($value, $interpolate = true)
 
             return sprintf('array(%s)', implode(', ', $code));
         } elseif ($value instanceof IteratorArgument) {
+            $countCode = array();
+            $countCode[] = 'function() {';
+            $operands = array(0);
+
             $code = array();
             $code[] = 'new RewindableGenerator(function() {';
-            foreach ($value->getValues() as $k => $v) {
-                $v = $this->wrapServiceConditionals($v, sprintf("        yield %s => %s;\n", $this->dumpValue($k, $interpolate), $this->dumpValue($v, $interpolate)));
+            foreach ($value->getValues() as $k => $value) {
+                $v = $this->wrapServiceConditionals($value, sprintf("        yield %s => %s;\n", $this->dumpValue($k, $interpolate), $this->dumpValue($value, $interpolate)));
                 foreach (explode("\n", $v) as $v) {
                     if ($v) {
                         $code[] = '    '.$v;
                     }
                 }
+
+                ($c = $this->getServiceConditionals($value)) ? $operands[] = "(int) ($c)" : ++$operands[0];
```

Comment:
there are two cases:
- two optional lines in the list of values: these should count from zero to two contionaly
- a single line with two contitional inside (eg - `[ @?foo, @?bar ]`: these count for zero or one only of course